src/Driver/PDO/Connection.php

```diff
@@ -87,17 +91,80 @@ public function quote(string $value): string
         return $this->connection->quote($value);
     }
 
-    public function lastInsertId(?string $name = null): string
+    /**
+     * {@inheritDoc}
+     */
+    public function lastInsertId()
     {
+        $driverName = $this->connection->getAttribute(PDO::ATTR_DRIVER_NAME);
+
+        if ($driverName === 'oci') {
+            throw IdentityColumnsNotSupported::new();
+        }
+
         try {
-            if ($name === null) {
-                return $this->connection->lastInsertId();
+            $lastInsertId = $this->connection->lastInsertId();
+        } catch (PDOException $exception) {
+            [$sqlState] = $exception->errorInfo;
+
+            // PDO PGSQL throws a 'lastval is not yet defined in this session' error when no identity value is
+            // available, with SQLSTATE 55000 'Object Not In Prerequisite State'
+            if ($sqlState === '55000' && $this->connection->getAttribute(PDO::ATTR_DRIVER_NAME) === 'pgsql') {
```

Comment:
This looks like logic for the `ExceptionConverter`.

Reply:
Please see [this comment](https://github.com/doctrine/dbal/pull/4324/files/83e4265f9d3172e507d6b33d06156fcc66dd6fea#r499680978).

`ExceptionConverter` lacks some context (what method was being called when the error occurred): a given error code may for example represent a Table not found, or a Sequence does not exist.

If you're not willing to introduce wrapper-level exception classes for identity columns / sequences -related exceptions, then we can just remove this code altogether and get away with it.